### ISSUE TITLE
Inconsistent keyboard behaviour in Search block

### ISSUE THREAD

**Issue by tellthemachines** [WordPress/gutenberg#38261]
### Description

When moving through the Search block with the arrow keys, focusing on the "Add label" field and then pressing Arrow Down takes us to the "Optional placeholder" field, and Arrow Down again takes us out of Search and into the next block. However, navigating back with Arrow Up key, the first field to receive focus in the Search block is the "Add button text" field, followed by the "Add label" field. 

On the way down, "Add button text" is skipped. On the way up,  "Optional placeholder" is skipped. 

Focus order should be the same in both directions.

### Step-by-step reproduction instructions

1. In the block editor, add a Search block, followed by a paragraph or other block
2. Starting with focus in the "Add label" field, move focus with Arrow Down until it leaves the Search block
3. Move focus back into Search with Arrow Up
4. Verify that focus order is different in both directions.

### Screenshots, screen recording, code snippet

[video recording](https://www.youtube.com/watch?v=au-hF14Zf3E)

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by JustinyAhin** [comment]
@tellthemachines this seems to have been fixed (I guess in previous Gutenberg releases).

https://user-images.githubusercontent.com/33403964/167603941-4ea492e3-8efe-4981-bbf3-70be4b3818ce.mp4
